gstreamer-video/src/video_codec_state.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2017 Thibault Saunier <tsaunier@gnome.org>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use ffi;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+use gst;
+use gst::miniobject::MiniObject;
+
+use video_info::VideoInfo;
+
+glib_wrapper! {
+    pub struct VideoCodecState(Shared<ffi::GstVideoCodecState>);
+
+    match fn {
+        ref => |ptr| ffi::gst_video_codec_state_ref(ptr),
+        unref => |ptr| ffi::gst_video_codec_state_unref(ptr),
+        get_type => || ffi::gst_video_codec_state_get_type(),
+    }
+}
+
+impl VideoCodecState {
```

Comment:
This API is not enough, subclasses need to be able to modify the fields of the codec state

Reply:
Done.

Reply:
Same thing as for the VideoCodecFrame. Some kind of locking would be necessary here, and I think the same approach would work for now.